docs/non_smooth.rst

```diff
@@ -22,12 +22,69 @@ Proximal gradient
 
     jaxopt.ProximalGradient
 
+Instantiating and running the solver
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Proximal gradient is a generalization of :ref:`projected gradient descent
+<constrained_optim>`. The non-smooth term :math:`g` above is specified by
+setting the corresponding ``prox`` argument.
+
+For instance, suppose we want to minimize
```

Comment:
```suggestion
For instance, suppose we want to solve the following optimization problem
```
(otherwise it can  be interpreted as minimizing the minimum)